### ISSUE TITLE
macOS Sierra

### ISSUE THREAD

**Issue by ghost** [dionmunk/ubersicht-network-throughput#1]
Since upgrading to macOS Sierra it no longer works. Any idea why that might be?

**Comment by dionmunk** [comment]
@Joooe10 I will have to look into this. I'll let you know as soon as I have something to report.

**Comment by Wes974** [comment]
So, any ideas ?

**Comment by dionmunk** [comment]
I will update today and let you know.

**Comment by dionmunk** [comment]
It would seem that the 'sar' command, the basis of this widget, has been removed in OS X Sierra. I am looking into an alternative method for getting network throughput.

**Comment by Wes974** [comment]
Yes I have seen this when I tried to run the code manually

**Comment by dionmunk** [comment]
Here is what I have come up with so far:

Using netstat I can get the current up/down speed (3rd and 6th numbers) for all network adapters with:

```
netstat -w1
```

The problem with this is that it is a command that runs forever on repeat. The `-w1` tells it to repeat ever second. It is the only way I have been able to find this output from netstat, or anywhere for that matter. Since the command runs forever, it isn't easy to capture the output to display in the widget. Using this repeating command is the only way I have found to get this data out of netstat.

Here are the other options I have tried at getting the above command pulled into something that I can work with.

The following will give us a line that begins with a number, so we only get the stuff we are looking for.

```
netstat -w1 | grep '[0-9].*'
```

The following will give us what we are looking for and then kill the `netstat -w1` process after it runs one time so we can use the output in the widget.

```
netstat -w1 | grep '[0-9].*' & sleep 1.5; kill $!
```

The problem here is that I can not parse the output after the process is _killed_. 

I have tried piping it to _awk_, but it returns nothing.

```
BYTES=$(netstat -w1 | grep '[0-9].*' & sleep 1.5; kill $!);
BYTES=$(echo $BYTES | awk '{print $3,$6}');
echo $BYTES;
```

This is where I am currently stuck.
